src/borg/archive.py

```diff
@@ -820,34 +820,55 @@ def stat_attrs(self, st, path):
         attrs.update(self.stat_ext_attrs(st, path))
         return attrs
 
-    def process_dir(self, path, st):
-        item = Item(path=make_path_safe(path))
-        item.update(self.stat_attrs(st, path))
+    @contextmanager
```

Comment:
IIRC @contextmanager is quite slow compared to implementing it directly (see backup_io changes). So maybe check/verify performance impact iff the PR should be merged and is ready for it.